jax/experimental/jet.py

```diff
@@ -32,6 +67,51 @@
 from jax import lax
 
 def jet(fun, primals, series):
+  r"""Taylor-mode higher-order automatic differentiation.
+
+  Args:
+    fun: Function to be differentiated. Its arguments should be arrays, scalars,
+      or standard Python containers of arrays or scalars. It should return an
+      array, scalar, or standard Python container of arrays or scalars.
+    primals: The primal values at which the Taylor approximation of ``fun`` should be
+      evaluated. Should be either a tuple or a list of arguments,
+      and its length should be equal to the number of positional parameters of
+      ``fun``.
+    series: Higher order Taylor-series-coefficients.
+      Together, `primals` and `series` make up a truncated Taylor polynomial.
+      Should be either a tuple or a list of tuples or lists,
+      and its length dictates the degree of the truncated Taylor polynomial.
+
+  Returns:
+    A ``(primals_out, series_out)`` pair, where ``primals_out`` is ``fun(*primals)``,
+    and together, ``primals_out`` and ``series_out`` are a
+    truncated Taylor polynomial of :math:`f(x(\cdot))`.
+    The ``primals_out`` value has the same Python tree structure as ``primals``,
+    and the ``series_out`` value the same Python tree structure as ``series``.
+
+  For example:
+
+  >>> import jax
+  >>> import jax.numpy as np
+
+  Consider the function :math:`x(t) = t^3`, :math:`t_0 = 0.5`,
+  and the first few Taylor coefficients
+  :math:`x_0=t_0^3`, :math:`x_1=3t_0^2`, and :math:`x_2=6t_0`.
+
+  >>> x0, x1, x2 = 0.5**3., 3.*0.5**2., 6.*0.5
+  >>> y0, (y1, y2) =  jet(np.sin, (x0,), ((x1, x2),))
+
+  :func:`jet` returns the Taylor coefficients of :math:`f(x(t)) = \sin(t^3)`:
+
+  >>> print(y0,  np.sin(x0))
+  0.12467473 0.12467473
+
+  >>> print(y1, np.cos(x0) * x1)
+  0.7441479 0.74414825
+
+  >>> print(y2, -np.sin(x0) * x1 +  x2)
+  2.9064622 2.906494
```

Comment:
In order to make, for instance, a `doctest` pass reliably, I would have to round these values. But I didnt see much rounding happening similar examples in other modules, and `pytest -n auto jax/experimental/jet.py` does not check these examples at least.